rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildConfigurationEndpointImpl.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    @Override
+    public BuildConfiguration getSpecific(int id) {
+        return super.getSpecific(id);
+    }
+
+    @Override
+    public void update(int id, BuildConfiguration buildConfiguration) {
+        super.update(id, buildConfiguration);
+    }
+
+    @Override
+    public void deleteSpecific(int id) {
+        super.delete(id);
+    }
+
+    @Override
+    public Build trigger(int id, BuildParameters buildParams, String callbackUrl) {
+        // TODO: implement
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    @Override
+    public Page<Build> getBuilds(int id, PageParameters pageParams, BuildsFilterParameters buildsFilter) {
+        // TODO: handle buildsFilter
+        return buildProvider.getBuildsForBuildConfiguration(
+                pageParams.getPageIndex(),
+                pageParams.getPageSize(),
+                pageParams.getSort(),
+                pageParams.getQ(),
+                id);
+    }
+
+    @Override
+    public BuildConfiguration clone(int id) {
+        return buildConfigurationProvider.clone(id);
+    }
+
+    @Override
+    public Page<GroupConfiguration> getGroupConfigs(int id, PageParameters pageParams) {
+
+        return groupConfigurationProvider.getGroupConfigurationsForBuildConfiguration(
+                pageParams.getPageIndex(),
+                pageParams.getPageSize(),
+                pageParams.getSort(),
+                pageParams.getQ(),
+                id);
+    }
+
+    @Override
+    public Page<BuildConfiguration> getDependencies(int id, PageParameters pageParams) {
+
+        return buildConfigurationProvider.getDependencies(
+                pageParams.getPageIndex(),
+                pageParams.getPageSize(),
+                pageParams.getSort(),
+                pageParams.getQ(),
+                id);
+    }
+
+    @Override
+    public void addDependency(int id, BuildConfigurationRef dependency) {
+        buildConfigurationProvider.addDependency(id, dependency.getId());
+    }
+
+    @Override
+    public void removeDependency(int id, int dependencyId) {
+        buildConfigurationProvider.removeDependency(id, dependencyId);
+    }
+
+    @Override
+    public Page<BuildConfigurationRevision> getRevisions(int id, PageParameters pageParams) {
+
+        return buildConfigurationProvider.getRevisions(
+                pageParams.getPageIndex(),
+                pageParams.getPageSize(),
+                id);
+    }
+
+    @Override
+    public BuildConfigurationRevision createRevision(int id, BuildConfiguration buildConfiguration) {
+        buildConfigurationProvider.update(id, buildConfiguration);
+
+        Optional<BuildConfigurationRevision> buildConfigurationAuditedRestOptional =
+                buildConfigurationProvider.getLatestAuditedMatchingBCRest(buildConfiguration);
+
+        if (buildConfigurationAuditedRestOptional.isPresent()) {
+            return buildConfigurationAuditedRestOptional.get();
```

Comment:
You can use `.orElseThrow` on optionals.
Maybe using IllegalStateException would make more sense?
And `buildConfigurationAuditedRestOptional` matches the old type name :P